source/reference/method/mkdir.txt

```diff
@@ -17,10 +17,41 @@ Description
 
    Creates a directory at the specified path. This method creates
    the entire path specified if the enclosing directory or
-   directories do not already exit.
+   directories do not already exit. The user running the
+   :program:`mongo` shell must have permission to create directories in
+   the specified path.
 
    This method is equivalent to :command:`mkdir -p` with BSD or GNU utilities.
 
    The :method:`mkdir()` method has the following parameter:
 
    .. include:: /includes/apiargs/method-mkdir-param.rst
+
+   .. versionadded:: 3.4
+
+   :method:`mkdir()` returns a document with
+   information about the result of the operation.
+
+   On success, :method:`mkdir()` returns the following:
+
+   .. code-block:: javascript
+
+      { "exists" : true, "created" : true }
+
+   If the directory at the specified path already exists,
+   :method:`mkdir()` returns the following:
+
+   .. code-block:: javascript
+
+      { "exists" : true, "created" : false }
+
+Example
+-------
+
+The following command creates a directory called ``foo`` in the
+shell's current working directory.
+
+.. code-block:: javascript
+
+   mkdir("foo")
+   { "exists" : true, "created" : true }
```

Comment:
We should separate out code they should run and the results.  Otherwise, the copyable code-block would include the results which would make pasting weird.